android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt

```diff
@@ -576,11 +580,32 @@ constructor(
       defaultRepository.delete(resourceType = resourceType, resourceId = resourceIdentifier)
     }
   }
+  suspend fun updateResourceLastUpdatedLinkedAsSubject(
+    questionnaireResponse: QuestionnaireResponse
+  ) {
+    if (questionnaireResponse.hasSubject() && questionnaireResponse.subject.hasReference()) {
+      val resourceId = questionnaireResponse.subject.reference.extractLogicalIdUuid()
+      val resourceType =
+        questionnaireResponse
+          .subject
+          .extractType()
+          .toString()
+          .resourceClassType()
+          .newInstance()
+          .resourceType
```

Comment:
doesn't the above give you the ResourceType?
```suggestion
```

Reply:
@pld do you mind elaborating what you meant by "above"?

Reply:
The piece of code ```questionnaireResponse
  .subject
  .extractType()``` returns a empty string which might hold `ResourceType.name` e.g `Patient` or `Group` as string .Since it's not an instance of `ResourceType` the conversion is necessary

Reply:
ah gotcha thanks